src/core/server.cc

```diff
@@ -109,6 +113,90 @@ InferenceServer::InferenceServer()
   inflight_request_counter_ = 0;
 }
 
+void
+InferenceServer::PrintSummary()
+{
+  // Models Summary
+  auto model_states = model_repository_manager_->BackendStates();
+
+  std::vector<std::string> model_headers;
+  model_headers.emplace_back("Model");
+  model_headers.emplace_back("Version");
+  model_headers.emplace_back("Status");
+
+  std::unique_ptr<triton::common::TablePrinter> models_table;
+  triton::common::TablePrinter::Create(&models_table, model_headers);
+
+  for (const auto& model_state : model_states) {
+    std::vector<std::string> model_record;
+    auto model_version_map = model_state.second;
+    std::string model_name = model_state.first;
+
+    // If model_version_map size is zero, no version is found for this model
+    if (model_version_map.size() == 0) {
+      model_record.emplace_back(model_name);
+      model_record.emplace_back("-");
+      model_record.emplace_back("Not loaded: No model version was found");
+      models_table->insert_row(model_record);
+    } else {
+      for (const auto& model_map : model_version_map) {
+        std::string model_version = std::to_string(model_map.first);
+        auto model_status_pair = model_map.second;
+        std::string model_status =
+            ModelReadyStateString(model_status_pair.first);
+
+        if (model_status_pair.second != "") {
+          model_status += ": " + model_status_pair.second;
+        }
+
+        model_record.emplace_back(model_name);
+        model_record.emplace_back(model_version);
+        model_record.emplace_back(model_status);
+        models_table->insert_row(model_record);
```

Comment:
You are re-using the same model_record for all versions in the model_version_map. Shift declaration of model record inside for loop.